src/App.tsx

```diff
@@ -1,223 +1,319 @@
+    setCompletedTodos(() => todos.filter(todo => todo.completed));
+  }, [todos]);
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+    if (!user) {
+      return;
+    }
+
+    if (newTodoTitle.trim().length === 0) {
+      setError(true);
+      setMessageError('Title can\'t be empty');
+
+      return;
+    }
+
+    setVisibelTodos(visibleTodos => [...visibleTodos, {
+      id: 0,
+      userId: user.id,
+      title: newTodoTitle,
+      completed: false,
+    }]);
+
+    try {
+      const isAdding = await createTodo({
+        userId: user.id,
+        title: newTodoTitle,
+        completed: false,
+      });
+
+      setTodos(PrevTodos => [...PrevTodos, isAdding]);
+    } catch (errorFromServer) {
+      setError(true);
+      setMessageError('Unable to add a todo');
+    } finally {
+      setVisibelTodos(visibleTodos => visibleTodos
+        .filter(todo => todo.id !== 0));
+    }
+
+    setTitle('');
+  };
+
+  const handleRemoveTodo = async (removeTodoID: number) => {
+    setActiveTodoId(idActive => [...idActive, removeTodoID]);
+
+    try {
+      await deleteTodo({ id: removeTodoID });
+
+      setTodos(Prevtodos => Prevtodos.filter(todo => todo.id !== removeTodoID));
+    } catch (errorFromServer) {
+      setError(true);
+      setMessageError('Unable to delete a todo');
+    } finally {
+      setActiveTodoId(idActive => idActive.filter(id => id !== removeTodoID));
+    }
+  };
+
+  const handleRemoveComleted = () => {
+    completedTodos.forEach((todoComleted) => handleRemoveTodo(todoComleted.id));
+  };
+
+  const handleUpdate = async (changeTodo: Todo) => {
+    if (!changeTodo.id) {
+      return;
+    }
+
+    try {
+      setActiveTodoId(idActive => [...idActive, changeTodo.id]);
+      await updateTodo(changeTodo.id, { completed: !changeTodo.completed });
+
+      setTodos(Prevtodos => [...Prevtodos].map(todo => {
+        if (todo.id === changeTodo.id) {
+          // eslint-disable-next-line no-param-reassign
+          todo.completed = !todo.completed;
+        }
+
+        return todo;
+      }));
+    } catch (errorFromServer) {
+      setError(true);
+      setMessageError('Unable to change a todo');
+    } finally {
+      setActiveTodoId(idActive => idActive.filter(id => id !== changeTodo.id));
+    }
+  };
+
+  const handleUpdateAll = () => {
+    todos.forEach(async changeTodo => {
+      try {
+        setActiveTodoId(idActive => [...idActive, changeTodo.id]);
+        await updateTodo(
+          changeTodo.id,
+          { completed: completedTodos.length !== todos.length },
+        );
+        setTodos(state => [...state].map(todo => {
+          const statusComleted = completedTodos.length !== todos.length;
+
+          // eslint-disable-next-line no-param-reassign
+          todo.completed = statusComleted;
+
+          return todo;
+        }));
+      } catch (errorFromServer) {
+        setError(true);
+        setMessageError('Unable to change a todo');
```

Comment:
also, you can create enum for error messages. It's a good practice not rely on the unknown string

Reply:
done